src/Mutator/NodeMutationGenerator.php

```diff
@@ -106,7 +107,14 @@ public function generate(Node $node): iterable
         }
 
         foreach ($this->mutators as $mutator) {
-            yield from $this->generateForMutator($node, $mutator);
+            foreach ($this->generateForMutator($node, $mutator) as $mutation) {
+                if ($this->mutantId !== null && $mutation->getHash() !== $this->mutantId) {
+                    continue;
+                }
+
+                // todo check that prepending code with same mutator doesn't invalidate ID
+                yield $mutation;
```

Comment:
A potentially better place for this is where we do all other filtering; also it feels like we are breaking the SRP here 🤔